cas-server-support-jdbc/src/main/java/org/jasig/cas/services/JpaServiceRegistryDaoImpl.java

```diff
@@ -35,6 +37,7 @@
     @PersistenceContext
     private EntityManager entityManager;
 
+    @Transactional(readOnly = false)
```

Comment:
Yes, that sounds good. I've always been expected to see `@Transactional` annotation at the registry level.

Though, in the `CentralAuthenticationServiceImpl` class, every annotated method generally performs a read then a write on a ticket. They were previously enclosed in the same transaction, I fear some side effects by having now separate transactions: what do you think?